src/index.jsx

```diff
@@ -49,27 +55,77 @@ function Form() {
         setPref('app', app);
     }, [app])
 
+    const menuItems = Object.entries(AVAILABLE_APPS).map(([name, value]) => ({
+        text: value.title,
+        key: name
+    }));
+    
+    const handleMenuChange = (item) => {
+        if (item) {
+            setApp(item.key);
+        }
+    };
+
+    const changeOpenState = (item) => {
+        if(isDropdownOpen == true){
+            document.getElementById("root").className = "normal";
+            document.getElementById("popoutBody").className = "normal";
+            setDropdownOpen(false);
+        } else {
+            document.getElementById("root").className = "expanded";
+            document.getElementById("popoutBody").className = "expanded";
+            setDropdownOpen(true);
+        }
+        
+    };
+
     return <Box display="flex" flexDirection="column">
+        {/* https://datahub.berkeley.edu */}
         <Heading sx={{ fontSize: 2, mb: 1 }}>JupyterHub URL</Heading>
 
-        <TextInput value={hubUrl} onChange={(ev) => setHubUrl(ev.target.value)} placeholder="https://myjupyterhub.org" aria-label="JupyterHub URL" />
+        <TextInput
+            value={hubUrl}
+            onChange = {
+                (ev) => setHubUrl(ev.target.value)
+            }
+            placeholder="https://myjupyterhub.org"
+            aria-label="JupyterHub URL"
+            sx={{ pt: 0.5, pb: 0.5 }}
+        />
+
         <Text color="danger.fg" sx={{ visibility: isValidHubUrl ? "hidden" : "visible" }}>Enter a valid URL</Text>
 
-        <Heading sx={{ fontSize: 2, mb: 1, mt: 2 }}>Open in</Heading>
-        <select className="form-select mb-1" onChange={(ev) => setApp(ev.target.value)} value={app}>
-            {Object.entries(AVAILABLE_APPS).map(([name, value]) => {
-                return <option key={name} value={name}>{value.title}</option>
-            })};
-        </select>
-
-        <Button disabled={!isValidHubUrl || finishedCopying} sx={{ mt: 2 }} onClick={() => {
-            copyGeneratedUrl(hubUrl, app);
-            // Flash a 'Copied!' message for 3 seconds after copying
-            setFinishedCopying(true);
-            setTimeout(() => setFinishedCopying(false), 3 * 1000)
-        }}>
-            <CopyIcon /> {finishedCopying ? "Copied!" : "Copy nbgitpuller link"}
-        </Button>
+        <Heading sx={{ fontSize: 2, mb: 1, mt: 3 }}>Open in</Heading>
+
+        <DropdownMenu
+            items={menuItems}
+            onChange={handleMenuChange}
+            placeholder={AVAILABLE_APPS[app].title} // Display the value of 'name' or a placeholder
+            open={isDropdownOpen}
+            onOpenChange={changeOpenState}
+        />
+
+        <div style={{ display: 'flex', gap: '8px', justifyContent: 'center' }}>
```

Comment:
In general, I think all CSS should just be in CSS classes rather than here as inline styles. Can you move those over and classes here (and in other places in the PR that use CSS)?

Reply:
removed all inline CSS